build.gradle.kts

```diff
@@ -56,15 +56,38 @@ allprojects {
 }
 
 subprojects {
-  apply(plugin = "com.diffplug.spotless")
-  configure<SpotlessExtension> {
-    setEnforceCheck(false)
-    kotlin {
-      target("**/*.kt")
-      ktlint(libs.versions.ktlint.get()).userData(kotlin.collections.mapOf("indent_size" to "2"))
-      trimTrailingWhitespace()
-      endWithNewline()
-      toggleOffOn()
+  if (name != "wire-golden-files") {
+    apply(plugin = "com.diffplug.spotless")
+    configure<SpotlessExtension> {
+      val licenseHeaderFile = rootProject.file("gradle/license-header.txt")
+      kotlin {
+        target("**/*.kt")
+        ktlint(libs.versions.ktlint.get()).editorConfigOverride(
+          mapOf("ktlint_standard_filename" to "disabled"),
+        )
+        trimTrailingWhitespace()
+        endWithNewline()
+        toggleOffOn()
+        // We cannot use it because the licensing job happens before. The header will be replaced and
+        // spotless will throw. We have to manually exclude stuff for now.
+        // See https://github.com/diffplug/spotless/discussions/1738
+        toggleOffOnRegex("""^(// Code generated by Wire protocol buffer compiler[\s\S]*)$""")
```

Comment:
Clever

Reply:
Would have been if it worked!

Reply:
https://github.com/diffplug/spotless/pull/1749

Reply:
Did you consider using excludes by path? Surely the generated code are all under individual folders and not mixed in with normal sources, right?

Reply:
Exclude by path already exist, right? And I'm actually using it because the thing I tried above didn't work. I'd want something to set in one place and never touch again. I don't know how Spotless will take my PR but that seems a better solution to me that manually adding paths